runtime/runtime/src/prefetch.rs

```diff
@@ -157,12 +164,12 @@ impl TriePrefetcher {
         self.prefetch_api.clear_data();
     }
 
-    fn prefetch_trie_key(&self, trie_key: TrieKey) -> Result<(), ()> {
+    fn prefetch_trie_key(&self, trie_key: TrieKey) -> Result<(), PrefetchQueueFullError> {
```

Comment:
I propose that the error type returned here be something like `enum PrefetchTrieKeyError` and `QueueFull` be a variant of it.  That makes it easier to add additional error types in future with minimal changes.

Reply:
LGTM!